src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -557,24 +568,42 @@ def augment_transactions_with_spans(
     # Fetch parent span ids of segment spans and their corresponding
     # transaction id so we can link parent/child transactions in
     # a trace.
-    spans_params = params.copy()
-    spans_params["project_objects"] = [p for p in params["project_objects"] if p.id in projects]
-    spans_params["project_id"] = list(projects.union(set(problem_project_map.keys())))
-
-    parents_results = SpansIndexedQueryBuilder(
-        Dataset.SpansIndexed,
-        spans_params,
-        query=f"trace:{trace_id} span_id:[{','.join(query_spans)}]",
-        selected_columns=[
-            "transaction.id",
-            "span_id",
-            "timestamp",
-        ],
-        orderby=["timestamp", "id"],
-        limit=10000,
-    ).run_query(referrer=Referrer.API_TRACE_VIEW_GET_PARENTS.value)
 
-    parent_map = {parent["span_id"]: parent for parent in parents_results["data"]}
+    def load_results(chunked_project_ids):
+        spans_params = params.copy()
+        spans_params["project_objects"] = [
+            p for p in params["project_objects"] if p.id in chunked_project_ids
+        ]
+        spans_params["project_id"] = chunked_project_ids
+        return SpansIndexedQueryBuilder(
+            Dataset.SpansIndexed,
+            spans_params,
+            query=f"trace:{trace_id} span_id:[{','.join(query_spans)}]",
+            selected_columns=[
+                "transaction.id",
+                "span_id",
+                "timestamp",
+            ],
+            orderby=["timestamp", "id"],
+            limit=10000,
+        ).run_query(referrer=Referrer.API_TRACE_VIEW_GET_PARENTS.value)
+
+    parent_map = {}
+    chunk_size = math.ceil(len(projects) / MAX_PARALLEL_QUERIES)
+    # Limit to 10 projects per query
+    if chunk_size > 10:
+        chunk_size = 10
+
+    with ThreadPoolExecutor(max_workers=MAX_PARALLEL_QUERIES) as executor:
+        futures = {
+            executor.submit(load_results, chunked_projects)
+            for chunked_projects in chunk(projects, chunk_size)
+        }
+        for future in as_completed(futures):
+            parents_results = future.result()
+            for parent in parents_results["data"]:
+                parent_map[parent["span_id"]] = parent
```

Comment:
hmmm sorta a nit, but I'd prefer we use the same `bulk_snql_query` here to run async